source/polyfill-selection.txt

```diff
@@ -1,63 +0,0 @@
-.. _cpp-polyfill-selection:
-
-=======================
-Choose a C++17 Polyfill
-=======================
-
-.. contents:: On this page
-   :local:
-   :backlinks: none
-   :depth: 1
-   :class: singlecol
-
-.. facet::
-   :name: genre
-   :values: reference
-
-The mongocxx driver uses C++17 features ``std::optional`` and ``std::string_view``.
-To use the C++17 standard library implementations for these features, set
-the CMake configuration variable ``CMAKE_CXX_STANDARD`` to 17 or higher.
-Otherwise, to compile the mongocxx driver for pre-C++17 configurations, a
-polyfill library implementation must be selected from the following options:
-
-- bsoncxx (*default only when -DENABLE_BSONCXX_POLY_USE_IMPLS=ON*)
-
-  Select with ``-DBSONCXX_POLY_USE_IMPLS=ON``. This option is most recommended, as
-  it does not require additional external library dependencies. To enable
-  selecting this option by default for pre-C++17 configurations when no other
-  options are specified, set ``ENABLE_BSONCXX_POLY_USE_IMPLS=ON`` (this option
-  will be set to ON by default in an upcoming major release).
-
-- MNMLSTC/core (*default for non-Windows platforms*)
-
-  **This option is deprecated and will be removed in an upcoming major release.**
-  Select with ``-DBSONCXX_POLY_USE_MNMLSTC=1``. 
-  
-  .. note:: 
-
-   This option vendors a header-only installation of MNMLSTC/core into the bsoncxx library installation and will therefore download MLNMLSTC from GitHub during the configuration process. If you already have an available version of MNMLSTC on your system, you can avoid the download step by using ``-DBSONCXX_POLY_USE_SYSTEM_MNMLSTC``.
-
-- Boost (*default for Windows platforms*)
-
-  **This option is deprecated and will be removed in an upcoming major release.**
-  Select with ``-DBSONCXX_POLY_USE_BOOST=1``. This is currently the only
-  non-bsoncxx option if you are using a version of MSVC that does not support
-  C++17.
-
-.. note::
-
-   "default" refers to **pre-C++17** configurations when no polyfill library is explicitly selected.
-
-Most users should use default behavior with ``-DENABLE_BSONCXX_POLY_USE_IMPLS=ON``.
-However, if you have a preference for one of the external polyfill libraries
-(e.g. already a dependency being used by your application), you may prefer to
-explicitly select that external polyfill library rather than rely on default
-selection behavior.
```

Comment:
Recommend restoring this page with the following contents (or perhaps inline the contents in the "Choose a C++17 Polyfill" section in `index.txt` and rename to "C++17 Polyfill Configuration"):

```
The MongoDB C++ Driver uses C++17 features ``std::optional<T>`` and
``std::string_view``. The bsoncxx library provides polyfill implementations for
these features when configured with a pre-C++17 standard. These bsoncxx polyfill
implementations will be used when ``CMAKE_CXX_STANDARD`` is less than ``17``
(set to `11` by default). To use the C++17 standard library implementations, set
``CMAKE_CXX_STANDARD`` to ``17`` during CMake configuration.
```

Reply:
Added to index.txt and incorporated the note / warning admonitions